marconi-sidechain/test/Spec/Marconi/Sidechain/Api/Query/Indexers/Utxo.hs

```diff
@@ -136,3 +184,70 @@ generateUtxoRowJson = do
                 (Just $ C.hashScript script)
         utxoRow = UtxoRow utxo (C.SlotNo 1) blockHeaderHash
     pure $ Aeson.encodePretty utxoRow
+
+-- | Create http JSON-RPC client function to test RPC route and handlers
+rpcHttpClient
+  :: String   -- ^ bech32 ShelleyEra address to fetch Utxo's for
+  -> (Warp.Port -> IO (JsonRpcResponse String UtxoQueryResult))
+rpcHttpClient address port = do
+  manager' <- newManager defaultManagerSettings
+  let env = mkClientEnv manager' (baseUrl port)
+  let (rpcEcho :<|> rpcTargets :<|> rpcUtxos) = mkHoistedHttpRpcClient env
+  -- test will fail if these calls/routes fail.
+  _ <- rpcEcho "client calling "
+  _ <- rpcTargets ""
+  rpcUtxos address
+
+-- | Test round trip Utxos thruough JSON-RPC http server
+-- We compare a represenation of the generated UtxoEvents
+-- with those fetched from the JSON-RPC  server. The purpose of this is:
+--   + RPC server routes the request to the correct handler
+--   + Events are serialized/deserialized thru the RPC layer and it various wrappers correctly
+
+
+propUtxoEventInsertionAndJsonRpcQueryRoundTrip
+  :: RpcClientAction
+  -> Property
+propUtxoEventInsertionAndJsonRpcQueryRoundTrip action = property $ do
+  (RpcEnv storableAction, rpcClient) <- liftIO action
+  events <- forAll genShelleyEraUtxoEvents
+  liftIO $ storableAction events
+  let (qAddresses :: [String]) =
+        Set.toList . Set.fromList . fmap (unpack . C.serialiseAddress)
+        . mapMaybe (addressAnyToShelley . Utxo._address)
+        . concatMap  (Set.toList . Utxo.ueUtxos)
+        $ events
+  rpcResponses <- liftIO $ for qAddresses rpcClient
+  let (inserted :: Set Utxo.UtxoRow) = Set.fromList . concatMap Utxo.eventsToRows $ events
+      (fetched :: Set Utxo.UtxoRow) =  Set.fromList . concatMap fromQueryResult $ rpcResponses
+  -- TODO
+  -- May be remove in favor of the `inserted === fetched` test, when the serialisation bug is fixed
+  when (any rpcQueryHasFailed rpcResponses ) Hedgehog.failure -- TODO remove me when we solve the test failer re ScriptData serialization difference^
```

Comment:
This is still failing even after the PR which fixed the ScriptData serialization?

Reply:
it works OK now, uncommented the `===` test